latex/Inhalt/grundlagen.tex

```diff
@@ -0,0 +1,193 @@
+Eine kurze Übersicht ist in \cref{tab:notation} auffindbar.
+
+\begin{table}[h]
+    \centering
+    \caption{Notation zur Beschreibung der Assembly-Instruktionen}
+    \label{tab:notation}
+    \begin{tabular}{l | l}
+        A & Wert im Akkumulator\\
+        B & Wert in Register B\\
+        BC & Wert in Registerpaar BC\\
+        $[HL]$ & Wert an Addresse HL\\
+        10X & 10 in Zahlensystem X (H: Hex, D: Dez, O/Q: Oct, B: Bin)
+    \end{tabular}
+\end{table}
+
+Zahlenliterale werden durch Suffixe den entsprechenden Zahlensystemen zugewiesen --- H für Hexadezimal (Basis 16), D für Dezimal (Basis 10), O/Q für Octal (Basis 8) und B für Binär (Basis 2).
+
+\subsubsection{Registerzugriff}
+
+\begin{table}[h]
+    \centering
+    \caption{Beispielhafte Befehle zum Registerzugriff}
+    \label{tab:mov}
+    \begin{tabular}{l | l}
+        \asm{MOV B, A} & Setze B auf Wert in A\\
+        \asm{MOV M, A} & Setze $[HL]$ auf A\\
+        \asm{MOV A, M} & Setze A auf $[HL]$\\
+        \asm{MVI B, 0FFH} & Setze B auf 0FFH\\
+        \asm{LXI B, 1234H} & Setze BC auf 1234H\\
+    \end{tabular}
+\end{table}
+
+\subsubsection{Verzweigungen}
+
+Bei Assembly gibt es grundlegend 2 Verzweigungstypen: Jumps und Calls. Bei Jumps handelt es sich um direkte Sprünge zu einer Addresse im Arbeitsspeicher. Die äquivalente in Programmiersprachen wie C ist der \rust{goto} Befehl.
+
+Calls hingegen entsprechen grob dem Funktionsaufruf aus higher-level Sprachen. Beim Aufruf einer Call-Instruktion wird die momentane Position im Code (der PC) auf dem Stack abgelegt, dies ist die sogenannte Return-Addresse, und anschließend ein Sprung zur angegebenen Addresse (zum Funktions-Code) ausgeführt. Die Return-Instruktion stellt das Ende einer Funktion dar, sie setzt den PC auf das oberste Element des Stacks, bei korrekter Verwendung ist dies die Return-Addresse.
+
+\begin{table}[h]
+    \centering
+    \caption{Beispielhafte Befehle für Verzweigungen}
+    \label{tab:jmp}
+    \begin{tabular}{l | l}
+        \asm{JMP 0FFH} & Setze PC auf 0FFH\\
+        \asm{JZ  0FFH} & Setze PC auf 0FFH falls Zero-Flag gesetzt ist \\
+        \asm{CALL 0FFH} & Speichere PC auf dem Stack und setze ihn auf FFH \\
+        \asm{RET} & Setze PC auf oberstes Stack Element (und entferne es) \\
+    \end{tabular}
+\end{table}
+
+\subsubsection{Arithmetik}
+
+Der Intel 8080 implementiert eine 8-Bit Arithmetik. Dafür werden die Bytes als Zahlen im Zweierkomplement interpretiert. Die Ergebnisse der Operationen werden hauptsächlich im Akkumulator gespeichert --- Ausnahmen sind \zB Inkrementierungen der Register. Viele der Operationen bearbeiten auch die Flaggen, um Informationen über das Ergebnis zurückzugeben (siehe \cref{sec:flags} für Details).
```

Comment:
The section referred to is called "Flags", I don't think we should use "Flaggen" then